linear_regression/linear_regression.py

```diff
@@ -0,0 +1,114 @@
+"""
+Linear regression is the most basic type of regression commonly used for
+predictive analysis. The idea is preety simple, we have a dataset and we have
+a feature's associated with it. The Features should be choose very cautiously
+as they determine, how much our model will be able to make future predictions.
+We try to set these Feature weights, over many iterations, so that they best
+fits our dataset. In this particular code, i had used a CSGO dataset (ADR vs
+Rating). The Regression equation for this problem will be the equation of
+staright line (y = mx + c), where y is rating , c (intercept) is constant,
+m(slope) is regression coefficient, x is the ADR. Our Features would be these
+two and try to best fit a line through dataset and estimate the parameters.
+"""
+
+import pandas as pd
+
+
+def collect_dataset():
+    """ Collect dataset of CSGO
+    The dataset contains ADR vs Rating of a Player
+    :return : dataset obtained from the link, as matrix
+    """
+    dataset = pd.read_csv('https://raw.githubusercontent.com/yashLadha/The_' +
+                          'Math_of_Intelligence/master/Week1/ADRvsRating.csv',
+                          delimiter=',')
+    dataset = dataset.as_matrix()
+    return dataset
+
+
+def run_steep_gradient_descent(data_, len_data_, alpha_, slope_, intercept_):
+    """ Run steep gradient descent and updates the Feature's
+    Feature1 is the slope_ and Feature 2 is intercept_
+    :param data_     : contains the dataset for gradient descent
+    :param len_data_ : length of the data_
+    :param alpha_          : Learning rate of the model
+    :param slope_          : Feature 1, slope of the line that best fits the dataset
+    :param intercept_      : Feature 2, intercept formed by the best fit line
+    ;param return          : Updated Feature's, using
+                       curr_features - alpha_ * gradient(w.r.t. feature)
+    """
+    n = len_data_
+    slope_local = 0.0
+    intercept_local = 0.0
+
+    for i in range(0, n):
+        x = data_[i, 0]
+        y = data_[i, 1]
+
+        # Partial derivative of cost function w.r.t feature 1
+        dm = -((2/n)*x*(y - (slope_ * x + intercept_)))
+
+        # Partial derivative of cost function w.r.t feature 2
+        dc = -((2/n)*(y - (slope_ * x + intercept_)))
+
+        slope_local += dm
+        intercept_local += dc
+
+    slope_updated = slope_ - alpha_ * slope_local
+    intercept_updated = intercept_ - alpha_ * intercept_local
+    return slope_updated, intercept_updated
+
+
+def sum_of_square_error(data_, len_data_, slope_, intercept_):
```

Comment:
I see you are suffixing an underscore after parameters and I don't see a good reason to do so.